app/services/workflow_service/property_assessment.rb

```diff
@@ -0,0 +1,68 @@
+module WorkflowService
+  class PropertyAssessment < BaseWorkflowService
+    def call
+      calculate_property
+      true
+    end
+
+    private
+
+    def calculate_property
+      @remaining_mortgage_allowance = Threshold.value_for(:property_maximum_mortgage_allowance, at: @submission_date)
+      calculate_additional_properties
+      calculate_main_dwelling(request_data: @particulars.request.applicant_capital.property.main_home,
+                              response_data: @particulars.response.details.capital.property.main_dwelling)
+    end
+
+    def calculate_additional_properties
+      @particulars.request.applicant_capital.property.additional_properties.each do |additional_property|
+        response_data = OpenStruct.new(AssessmentParticulars.initial_property_details)
+        calculate_individual_property(request_data: additional_property, response_data: response_data, property_type: :additional_property)
+        @particulars.response.details.capital.property.additional_properties << response_data
+      end
+    end
+
+    def calculate_main_dwelling(request_data:, response_data:)
+      calculate_individual_property(request_data: request_data, response_data: response_data, property_type: :main_dwelling)
+    end
+
+    def calculate_individual_property(request_data:, response_data:, property_type:) # rubocop:disable Metrics/AbcSize
```

Comment:
Ultimately, this could be extracted to its own service

Reply:
Well we have a property service which handles both main dwelling and an array of additional properties.  The calculations are the same for both, except for a different disregard amount, so I think all in the same place is probably right.